File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        temp=deque()
        result=[]
        for part in s:
            if part=="(":
                temp.append(len(result))
            elif part==")":
                begin=temp.pop()
                result[begin:]=result[begin:][::-1]
            else:
                result.append(part)
        return "".join(result)

                

        
        




        

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    string reverseParentheses(string s) {
        stack<int> temp;
        string answer;
        for(char part:s){
            if (part=='('){
                temp.push(answer.length());
            }
            else if(part==')'){
                int start=temp.top();
                temp.pop();
                reverse(answer.begin()+start,answer.end());
            }
            else{
                answer+=part;
            }
        }
        return answer;

    }
};